src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -1205,7 +1205,10 @@ static void mmap_attach_shared(int vmid, char** addr, size_t* sizep, TRAPS) {
   FREE_C_HEAP_ARRAY(char, dirname);
   FREE_C_HEAP_ARRAY(char, filename);
 
-  if (fd == OS_ERR || HAS_PENDING_EXCEPTION) {
+  if (HAS_PENDING_EXCEPTION) {
+    assert(fd == OS_ERR, "open_sharedmem_file always return OS_ERR on exceptions");
```

Comment:
Does this really appease the static analysis tool?

Reply:
The static analysis tool should be fine about the above change. As I mentioned in the bug report, the warning is regarding the fd is not closed under the conditions HAS_PENDING_EXCEPTION and fd != OS_ERR.
I will run rerun the test to be sure.

Reply:
I re-ran tiers 1 and 2 tests using mach5 which tests debug build.
The run with the static analysis tool uses product build.
Both passed.